cmd/cli/vcancel/main.go

```diff
@@ -31,7 +31,7 @@ func main() {
 		Short: "cancel a job",
 		Long:  `cancel a running, pending, or aborted job with specified name in default or specified namespace`,
 		Run: func(cmd *cobra.Command, args []string) {
-			util.CheckError(cmd, vcancel.CancelJob())
+			util.CheckError(cmd, vcancel.CancelJob(cmd.Context()))
```

Comment:
Will `cmd.Context()` be nil?

Reply:
It will not be `nil`, it will be initialized by default when main.go is started.
The call chain is as follows

```go
func main() {
	rootCmd := cobra.Command{
		Use: "vcctl",
	}

	// tell Cobra not to provide the default completion command
	rootCmd.CompletionOptions.DisableDefaultCmd = true

	rootCmd.AddCommand(buildJobCmd())
	rootCmd.AddCommand(buildQueueCmd())
	rootCmd.AddCommand(versionCommand())

	code := cli.Run(&rootCmd)
	os.Exit(code)
}

func Run(cmd *cobra.Command) int {
	if logsInitialized, err := run(cmd); err != nil {
       ....
}

func run(cmd *cobra.Command) (logsInitialized bool, err error) {
	rand.Seed(time.Now().UnixNano())
	defer logs.FlushLogs()

	...

	err = cmd.Execute()
	return
}

// Execute uses the args (os.Args[1:] by default)
// and run through the command tree finding appropriate matches
// for commands and then corresponding flags.
func (c *Command) Execute() error {
	_, err := c.ExecuteC()
	return err
}

// ExecuteC executes the command.
func (c *Command) ExecuteC() (cmd *Command, err error) {
	if c.ctx == nil {
		c.ctx = context.Background()
	}

	// Regardless of what command execute is called on, run on Root only
	if c.HasParent() {
		return c.Root().ExecuteC()
	}

	// windows hook
	if preExecHookFn != nil {
		preExecHookFn(c)
	}
      .....
```

Reply:
Ok